iceoryx2/src/service/builder/publish_subscribe.rs

```diff
@@ -37,12 +38,12 @@ use self::{
 use super::{OpenDynamicStorageFailure, ServiceState};
 
 #[repr(C)]
-#[derive(Debug)]
+#[derive(Debug, ZeroCopySend)]
 #[doc(hidden)]
 pub struct CustomHeaderMarker {}
 
 #[repr(C)]
-#[derive(Debug)]
+#[derive(Debug, ZeroCopySend)]
```

Comment:
In line 309 in this file you have to restrict the `user_header` generic parameter `M` also to `ZeroCopySend`.

Reply:
Good catch. I overlooked more places in this file where `ZeroCopySend` is needed.